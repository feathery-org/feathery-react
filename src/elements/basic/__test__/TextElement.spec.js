import React from 'react';
import { create, act } from 'react-test-renderer';
import Elements from '../..';

describe('Text', () => {
  it('renders an empty text block', async () => {
    // Arrange
    const props = {
      element: {
        styles: {},
        mobile_styles: {},
        properties: { text_formatted: [{ insert: '\n' }] }
      },
      values: {},
      conditions: [],
      submit: () => {}
    };

    // Act
    let text;
    act(() => {
      text = create(<Elements.TextElement {...props} />);
    });
    const tree = text.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  it('renders a plaintext text block', async () => {
    // Arrange
    const props = {
      element: {
        styles: {},
        mobile_styles: {},
        properties: { text_formatted: [{ insert: 'Hello World!' }] }
      },
      values: {},
      conditions: [],
      submit: () => {}
    };

    // Act
    let text;
    act(() => {
      text = create(<Elements.TextElement {...props} />);
    });
    const tree = text.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  it('renders a text block with custom styles', async () => {
    // Arrange
    const props = {
      element: {
        styles: {},
        mobile_styles: {},
        properties: {
          text_formatted: [
            {
              insert: 'Hello World!',
              attributes: {
                weight: 700,
                color: '#000000FF',
                size: '48px',
                font: 'sans-serif',
                italic: true,
                strike: false,
                underline: false
              }
            }
          ]
        }
      },
      values: {},
      conditions: [],
      submit: () => {}
    };

    // Act
    let text;
    act(() => {
      text = create(<Elements.TextElement {...props} />);
    });
    const tree = text.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
