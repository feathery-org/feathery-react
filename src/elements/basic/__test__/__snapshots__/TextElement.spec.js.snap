// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders a plaintext text block 1`] = `
<div
  className="css-sa3tol-TextElement"
>
  <div
    className="css-795pei-TextElement"
  >
    <span
      className="css-1kxrhf3"
      onClick={[Function]}
    >
      Hello World!
    </span>
  </div>
</div>
`;

exports[`Text renders a text block with custom styles 1`] = `
<div
  className="css-sa3tol-TextElement"
>
  <div
    className="css-795pei-TextElement"
  >
    <span
      className="css-1kxrhf3"
      onClick={[Function]}
    >
      Hello World!
    </span>
  </div>
</div>
`;

exports[`Text renders an empty text block 1`] = `
<div
  className="css-sa3tol-TextElement"
>
  <div
    className="css-795pei-TextElement"
  >
    <span
      className="css-1kxrhf3"
      onClick={[Function]}
    >
      

    </span>
  </div>
</div>
`;
